### ISSUE TITLE
Bug with customer bills list

### ISSUE THREAD

**Issue by lelex86** [Dolibarr/dolibarr#26145]
### Bug

Hello everyone, 
I have noticed a bug with the customer invoice list. Indeed, if the 'third-party type' field is displayed, it becomes impossible to filter the list because the 'customer code' field automatically filters the third-party type ID if you filter by 'third-party type' or -1 for all other filters. If you hide the 'third-party type' field, then the list works normally.

### Environment Version

18.0.1

### Environment OS

Linux vps736401 5.15.0-86-generic #96-Ubuntu SMP Wed Sep 20 08:23:49 UTC 2023 x86_64

### Environment Web server

nginx/1.25.2

### Environment PHP

7.4.33

### Expected and actual behavior

When we want to filter the customer invoice, the list should filter.

### Steps to reproduce the behavior

Select the field "Third-party type" to be visible, then try to filter with any field.

### Attached files

![capture facture 1](https://github.com/Dolibarr/dolibarr/assets/75493227/519d3349-5a1b-4e52-a9ab-95b06e6eb673)
![capture facture 2](https://github.com/Dolibarr/dolibarr/assets/75493227/c8cbb27b-53b2-4006-b218-9b54eb312d9d)

**Comment by ksar-ksar** [comment]
Hello,

I confirm that, when you have selected a thirdparty type, for an unknown raison, it is also used in code-client

The POST variables are OK : 
```
array:39 [
  "token" => "b513da85d35ae525b1e5f5e22affe71e"
  "formfilteraction" => "list"
  "action" => "list"
  "sortfield" => "f.datef"
  "sortorder" => "DESC"
  "search_status" => "-1"
  "contextpage" => "invoicelist"
  "socid" => "0"
  "mode" => ""
  "massaction" => "0"
  "pageplusoneold" => "1"
  "limit" => "25"
  "search_sale" => "-1"
  "search_user" => "-1"
  "search_product_category" => "-1"
  "search_categ_cus" => "-1"
  "button_search_x" => "x"
  "search_ref" => ""
  "search_date_start" => ""
  "search_date_startday" => ""
  "search_date_startmonth" => ""
  "search_date_startyear" => ""
  "search_date_end" => ""
  "search_date_endday" => ""
  "search_date_endmonth" => ""
  "search_date_endyear" => ""
  "search_datelimit_end" => ""
  "search_datelimit_endday" => ""
  "search_datelimit_endmonth" => ""
  "search_datelimit_endyear" => ""
  "search_project_ref" => ""
  "search_company" => ""
  "search_company_alias" => ""
  "search_zip" => ""
  "search_type_thirdparty" => "7"
  "search_montant_ht" => ""
  "search_montant_ttc" => ""
  "search_login" => ""
  "selectedfields" => "f.ref,f.datef,f.date_lim_reglement,p.ref,s.nom,s.name_alias,s.zip,typent.code,f.total_ht,f.total_ttc,u.login,f.fk_statut,"
]

```
BUt then the SQL request is wrong with `s.code_client LIKE '%7%'`

`SELECT f.rowid as id, f.ref, f.ref_client, f.fk_soc, f.type, f.note_private, f.note_public, f.increment, f.fk_mode_reglement, f.fk_cond_reglement, f.total_ht, f.total_tva, f.total_ttc, f.localtax1 as total_localtax1, f.localtax2 as total_localtax2, f.fk_user_author, f.fk_multicurrency, f.multicurrency_code, f.multicurrency_tx, f.multicurrency_total_ht, f.multicurrency_total_tva as multicurrency_total_vat, f.multicurrency_total_ttc, f.datef, f.date_valid, f.date_lim_reglement as datelimite, f.module_source, f.pos_source, f.paye as paye, f.fk_statut, f.close_code, f.datec as date_creation, f.tms as date_update, f.date_closing as date_closing, f.retained_warranty, f.retained_warranty_date_limit, f.situation_final, f.situation_cycle_ref, f.situation_counter, s.rowid as socid, s.nom as name, s.name_alias as alias, s.email, s.phone, s.fax, s.address, s.town, s.zip, s.fk_pays, s.client, s.fournisseur, s.code_client, s.code_fournisseur, s.code_compta as code_compta_client, s.code_compta_fournisseur, s.parent as fk_parent, s2.nom as name2, typent.code as typent_code, state.code_departement as state_code, state.nom as state_name, country.code as country_code, f.fk_fac_rec_source, p.rowid as project_id, p.ref as project_ref, p.title as project_label, u.login, u.lastname, u.firstname, u.email as user_email, u.statut as user_statut, u.entity, u.photo, u.office_phone, u.office_fax, u.user_mobile, u.job, u.gender, ef.commission as options_commission, ef.reseller as options_reseller, ef.delayautopayment as options_delayautopayment FROM llx_societe as s LEFT JOIN llx_societe as s2 ON s2.rowid = s.parent LEFT JOIN llx_c_country as country on (country.rowid = s.fk_pays) LEFT JOIN llx_c_typent as typent on (typent.id = s.fk_typent) LEFT JOIN llx_c_departements as state on (state.rowid = s.fk_departement), llx_facture as f LEFT JOIN llx_facture_extrafields as ef on (f.rowid = ef.fk_object) LEFT JOIN llx_projet as p ON p.rowid = f.fk_projet LEFT JOIN llx_user AS u ON f.fk_user_author = u.rowid WHERE f.fk_soc = s.rowid AND f.entity IN (1) AND (s.code_client LIKE '%7%') AND s.fk_typent IN (7) ORDER BY f.datef DESC LIMIT 26 0.97`

**Comment by sonikf** [comment]
Hi @ksar-ksar 
This bug is fixed in 18.02 with #25955 please close.
